api/helpers/lock-game.js

```diff
@@ -0,0 +1,63 @@
+const { randomUUID } = require('crypto'); // Added in: node v14.17.0
+const dayjs = require('dayjs');
+
+async function sleep(durationInMillis) {
+  return new Promise((resolve) => {
+    setTimeout(resolve, durationInMillis);
+  });
+}
+
+module.exports = {
+  friendlyName: 'Lock a game',
+
+  description:
+    'Lock a game to reserve it for updates until the game is unlocked or the lock times out. Will error if we cannot obtain a lock before hitting the max wait time or the max number of attempts',
+
+  inputs: {
+    gameId: {
+      type: 'number',
+      descritiption: 'The database id of game to lock',
+      example: 44,
+      required: true,
+    },
+  },
+
+  fn: async ({ gameId }, exits) => {
+    const LOCK_RETRY_TIME_MS = 250;
+    const LOCK_MAX_WAIT_TIME_MS = 5000;
+    const MAX_ATTEMPTS = 20;
```

Comment:
It's kindof redundant to have both a max attempts count and a max wait time when we have a fixed interval retry time. As it stands, we'll retry every 250ms for up to 5s or 20 tries. Theoretically those are equivalent, but each attempt will take nonzero time so in reality only the 5s timeout would apply. 

So we could remove either the max wait time or the attempt count to simplify this because they can't both be used at once meaningfully. 

Alternatively, we could adjust the wait time to be non-fixed, perhaps exponentially increasing up to a maximum. That's what ButterBothSides (who recommended this overall approach) does at his work. This decreases the load on the database and increases the chance for successfully obtaining a lock when multiple simultaneous requests are processed. Having both max attempts and max wait time would allow for fine tuning of these params as we see fit.

In practice though for a given game, we're likely looking at a maximum of 3 simultaneous requests at once, so I suspect that exponential decay is overkill. That means we're likely best off using either max wait time, or max attempts. In that I'd say that max attempts is easier to read and understand b/c it's an incremented counter, and max wait time would arguably have the UX benefit of having a hard cap on the user wait time before their request is fully processed. @seriouslysean penny for your thoughts.

My inkling is to go for max attempts, in which case we could ditch the while loop for a for loop which will probably be easier to read